openedx/core/djangoapps/agreements/api.py

```diff
@@ -68,3 +73,180 @@ def get_integrity_signatures_for_course(course_id):
+    try:
+        course_ltipiitools = LTIPIITool.objects.get(course_key=course_key).lti_tools
+        return LTIToolsReceivingPIIData(
+            lii_tools_receiving_pii=course_ltipiitools,
+        )
+    except ObjectDoesNotExist:
+        return None
+
+
+def user_lti_pii_signature_needed(username, course_id):
+    """
+    Determines if a user needs to acknowledge the LTI PII Agreement
+
+    Arguments:
+        * username (str)
+
+    Returns:
+        * True if the user needs to sign a new acknowledgement.
+        * False if the acknowledgements are up to date.
+    """
+    if _course_has_lti_pii_tools(course_id):
+        if _user_lti_pii_signature_exists(username, course_id):
+            if _user_needs_signature_update(username, course_id):
+                # up to date
+                return False
+            else:
+                # lti pii signature needs to be updated
+                return True
+        else:
+            # write a new lti pii signature
+            return True
+    else:
+        return False
+
+
+def _course_has_lti_pii_tools(course_id):
+    """
+    Determines if a specifc course has lti tools sharing pii
+
+    Arguments:
+        * course_id (str)
+
+    Returns:
+        * True if the course does have a list.
+        * False if the course does not.
+    """
+    course_key = CourseKey.from_string(course_id)
+    try:
+        LTIPIITool.objects.get(course_key)
+    except ObjectDoesNotExist:
+        # no entry in the database
+        return False
+    else:
+        return True
+
+
+def _user_lti_pii_signature_exists(username, course_id):
+    """
+    Determines if a user's lti pii signature exists for a specfic course
+
+    Arguments:
+        * username (str)
+        * course_id (str)
+
+    Returns:
+        * True if user has a signature for the given course.
+        * False if the user does not have a signature for the given course.
+    """
+    course_key = CourseKey.from_string(course_id)
+
+    try:
+        user = User.objects.get(username=username)
+        LTIPIISignature.objects.get(user=user, course_key=course_key)
+    except ObjectDoesNotExist:
+        return None
+    else:
+        return True  # signature exist
+
+
+def _user_needs_signature_update(username, course_id):
+    """
+        Determines if a user's existing lti pii signature is out-of-date for a given course.
+
+        Arguments:
+            * username (str)
+            * course_id (str)
+
+        Returns:
+            * True if user needs a new signature
+            * False if the user has an up-to-date signature.
+        """
+    course_key = CourseKey.from_string(course_id)
+
+    try:
+        user = User.objects.get(username=username)
+        user_lti_pii_signature_hash = LTIPIISignature.objects.get(course_key=course_key, user=user).lti_tools_hash
+        course_lti_pii_tools_hash = LTIPIITool.objects.get(course_key=course_key).lti_tools_hash
+    except ObjectDoesNotExist:
```

Comment:
You can specify a set of exceptions to handle here, like `User.DoesNotExist`, `LTIPIISignature.DoesNotExist`, and `LTIPIITool.DoesNotExist` instead of the generic `ObjectDoesNotExist` exception; see my comment above.